game/test/graphic/DungeonCameraTest.java

```diff
@@ -1,135 +1,230 @@
-         Mockito.verify(frustum_noFollow).boundsInFrustum(any());
-         Mockito.verifyNoMoreInteractions(cam_noFollow, pos_noFollow, frustum_noFollow);
-     }
-
-     @Test
-     public void test_getFrustum() {
-         Mockito.verifyNoMoreInteractions(cam, pos, frustum, entity);
-     }
-
-     @Test
-     public void test_getFrustum_noFollow() {
-         Mockito.verifyNoMoreInteractions(cam_noFollow, pos_noFollow, frustum_noFollow);
-     }
-    */
+
+    @BeforeClass
+    public static void setUpGdx() {
+        GdxNativesLoader.load(); // load natives for headless testing
+    }
+
+    @Test
+    public void test_update() {
+        // Prepare
+        Entity entity = new Entity();
+        PositionComponent positionComponent = new PositionComponent(entity, new Point(2, 2));
+        DungeonCamera camera =
+                new DungeonCamera(
+                        positionComponent, Constants.VIEWPORT_WIDTH, Constants.VIEWPORT_HEIGHT);
+
+        // Testing
+        assertEquals(
+                "Pos should be (0, 0, 0) initially.",
+                camera.position,
+                new Vector3(0, 0, 0)); // Camera is initially at (0, 0, 0)
+        camera.update();
+        assertEquals(
+                "Pos should be (2, 2, 0) after update.",
+                camera.position,
+                new Vector3(2, 2, 0)); // Because it follows the positionComponent
+        positionComponent.setPosition(new Point(15, 23));
+        camera.update();
+        assertEquals(
+                "Pos should be (15, 23, 0) after update.",
+                camera.position,
+                new Vector3(15, 23, 0)); // Because it follows the positionComponent
+
+        // Cleanup
+        Game.getEntities().clear();
+        Game.getEntitiesToRemove().clear();
+    }
+
+    @Test
+    public void test_update_noFollow() {
+        // Prepare
+        DungeonCamera camera =
+                new DungeonCamera(null, Constants.VIEWPORT_WIDTH, Constants.VIEWPORT_HEIGHT);
+
+        // Testing
+        camera.update();
+        assertEquals("Pos should be (0, 0, 0) initially.", camera.position, new Vector3(0, 0, 0));
+        camera.setFocusPoint(new Point(15, 23));
+        assertEquals(
+                "Pos should be (0, 0, 0) before update.", camera.position, new Vector3(0, 0, 0));
+        camera.update();
+        assertEquals(
+                "Pos should be (15, 23, 0) after update.", camera.position, new Vector3(15, 23, 0));
+
+        // Cleanup
+        Game.getEntities().clear();
+        Game.getEntitiesToRemove().clear();
+    }
+
+    @Test
+    public void test_follow() {
+        // Prepare
+        Entity entity = new Entity();
+        PositionComponent positionComponent = new PositionComponent(entity, new Point(2, 2));
+        DungeonCamera camera =
+                new DungeonCamera(null, Constants.VIEWPORT_WIDTH, Constants.VIEWPORT_HEIGHT);
+
+        // Testing
+        assertEquals(
```

Comment:
java.lang.AssertionError: Camera should be at (0, 0, 0) initially. expected:<(1.0,0.0,0.0)> but was:<(0.0,0.0,0.0)>
so würde es um log stehen wenn es falsch wäre weis nicht ob das (0,0,0) notwendig wäre

Reply:
Stimmt.